src/main/java/org/folio/innreach/external/service/impl/InnReachAuthServiceImpl.java

```diff
@@ -0,0 +1,48 @@
+package org.folio.innreach.external.service.impl;
+
+import java.net.URI;
+import java.util.Base64;
+
+import com.google.common.cache.Cache;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import org.springframework.stereotype.Service;
+
+import org.folio.innreach.external.client.feign.InnReachAuthClient;
+import org.folio.innreach.external.dto.AccessTokenDTO;
+import org.folio.innreach.external.dto.AccessTokenRequestDTO;
+import org.folio.innreach.external.service.InnReachAuthService;
+
+@RequiredArgsConstructor
+@Log4j2
+@Service
+public class InnReachAuthServiceImpl implements InnReachAuthService {
+
+  private final InnReachAuthClient innReachAuthClient;
+  private final Cache<String, AccessTokenDTO> accessTokenCache;
+
+  @Override
+  public AccessTokenDTO getAccessToken(AccessTokenRequestDTO tokenRequestDTO) {
+    var cachedAccessToken = accessTokenCache.getIfPresent(tokenRequestDTO.getCentralServerUri());
```

Comment:
Would it be better to use Spring abstractions for caching like  @Cacheable with Caffeine as a provider?
Relevant documentation:
* https://docs.spring.io/spring-boot/docs/current/reference/html/features.html#features.caching.provider
* https://docs.spring.io/spring-framework/docs/current/reference/html/integration.html#cache

Later on we can apply the same technic to other external calls like  querying mod-users

Reply:
Yes you are right. I was thinking about Spring Cache but decided to use Guava as a temporary solution and discuss Spring cache with you later.